src/VisualStudio/Core/Def/Venus/ContainedDocument.cs

```diff
@@ -184,14 +185,33 @@ public DocumentId FindProjectDocumentIdWithItemId(uint itemidInsertionPoint)
                 }
             }
 
+            // The requested insertion itemid wasn't found. Check if the containing
+            //   document matches the requested itemid.
+            var containingDocumentItemid = GetContainingDocumentItemId(hierarchy);
```

Comment:
Was there a specific reason you put this second, or just generally being cautious against breaking anything that expects that earlier loop to be used?

Reply:
I was trying to be cautious, but it seems like a fairly safe change and it would be nice to short-circuit walking the project's documents.